src/snackbarApp/Snack.java

```diff
@@ -0,0 +1,102 @@
+package snackbarApp;
+
+public class Snack
+{
+	// fields first - state
+	private static int maxId = 0;
+	private int id;
+	private String name;
+	private double quantity; // real number - float
+	private double cost; // real number - float	
+	private int vendingMachineId;
+
+	// constructor - same name as the class, no return type. Already Void
+	// initial state of the object - current initial value of all the fields 
+	public Snack(String name, double quantity, double cost, int vendingMachineId)
+	{
+		maxId++;
+
+		id = maxId;
+
+		this.name = name;
+		this.quantity = quantity;
+		this.cost = cost;
+		this.vendingMachineId = vendingMachineId;
+	}
+
+	// methods - getters and setters
+	public void setId(int id)
+	{
+		this.id = id;
+	}
+
+	public int getId()
+	{
+		return id;
+	}
+
+	public void setName(String name)
+	{
+		this.name = name;
+	}
+
+	public String getName() // getters conventionally have 'get' and camelcase
+	{
+		return name;
+	}
+
+    public void setCost(Double cost)
+    {
+        this.cost = cost;
+    }
+
+    public double getCost()
+    {
+        return cost;
+    }
+
+    public int getVendingMachineId()
+    {
+        return vendingMachineId;
+    }
+
+    public void setVendingMachineId(int id)
+    {
+        this.vendingMachineId = id;
+    }
+
+   	public double getQuantity()
+	{
+		return quantity;
+	}
+
+	// other methods
+	public double setQuantity(double quantity)
+	{	
+		return this.quantity = quantity;
+	}
+
+	public double buySnack(double quantity)
+	{
+		setQuantity(this.quantity - quantity);
+		return cost * quantity;
+	}
+
+	public Double getTotal(Double quantity)
+	{
+		return quantity * cost;
+	}
```

Comment:
Use the primitive type `double` instead of `Double`. When you use the capitalized version you are calling a wrapper class for the primitive type.  The class has a bunch of methods that you don't need here and it uses a lot more memory so just use the primitive type.   Also instead of passing quantity of type `double`, you should make it type `int` instead since quantity will never not be a whole number.

```
public double getTotal(int quantity)
{
  return quantity * cost;
}
```